src/pp-ga/core.ts

```diff
@@ -0,0 +1,41 @@
+import './ga.js';
+import FieldsObject = UniversalAnalytics.FieldsObject;
+import packageConf from '../../package.json';
+
+const GA_ID_PROD = 'UA-123054125-1';
+const GA_ID_DEV = 'UA-123054125-2';
+
+export const GACustomFieldsIndex = {
+  eventUrl: 'dimension1',
```

Comment:
Is it that the data should be presented to GA using these "dimension" labels?

Reply:
GA has default dimensions (which are plenty) but are either basic and related to device, time etc or are more specialized but designed mainly for e-commerce. While we naturally, out-of-the-box use the first, we miss some more specific ones for our own purposes and here come _Custom Dimensions_ which are the way to extend default ones. In some other analytics system, in most cases paid, you do not have to declare like this, but that's how it goes with GA:
>Before you can send custom dimension and metric values to Analytics, they must first be defined in an Analytics property. Each Analytics property has 20 available indices for custom dimensions, and another 20 indices available for custom metrics.
>
>When you define a custom dimension or metric, you specify its name and other configuration values, and Analytics assigns an index number that you use later to reference the dimension or metric.

From https://support.google.com/analytics/answer/2709828?hl=en
Here is the list of the default dimensions: https://developers.google.com/analytics/devguides/reporting/core/dimsmets

Reply:
One more thing is that GA requires events to have `category` and `action` dimensions, while `label` is optional, in fact `label` is really what matters and most systems have only this single dimension and __`label` is what we are used to call _event___.